### PR TITLE
Setup registry.conf Env Vars for Buildah

### PR DESCRIPTION
**Pull request by adambkaplan**
* Remove build-registry-conf mount
* Supply ConfigMap for build-system-configs mount
* Mount build-system-configs for all build containers
* Rename registry config -> build system config

JIRA-ID: [DEVEXP-154](https://jira.coreos.com/browse/DEVEXP-154)

### CODE REVIEW

File: pkg/build/controller/strategy/util.go [hunk 1]
```diff
@@ -350,38 +348,61 @@ func setupContainersConfigs(pod *corev1.Pod, container *corev1.Container) {
...
+	// registries.conf is the primary registry config file mounted in by OpenShift
 	registriesConfPath := filepath.Join(configDir, "registries.conf")
+	// registries.d is a directory used by buildah to support multiple registries.conf files
+	// currently not created/managed by OpenShift
 	registriesDirPath := filepath.Join(configDir, "registries.d")
+	// signature-policy.json sets image signing policies for buildah (allowed signatures, etc.)
+	// currently not created/managed by OpenShift
 	signaturePolicyPath := filepath.Join(configDir, "signature-policy.json")
+	// storage.conf configures storage policies for buildah
+	// currently not created/managed by OpenShift
```
Comment (adambkaplan) [inline pkg/build/controller/strategy/util.go]: doc the relevant paths that are passed in as buildah env vars.

File: pkg/build/controller/strategy/util.go [hunk 2]
```diff
@@ -491,40 +512,3 @@ func setupBuildCAs(build *buildv1.Build, pod *corev1.Pod, includeAdditionalCA bo
 		pod.Spec.Containers = containers
 	}
 }
-
-func setupRegistries(build *buildv1.Build, pod *corev1.Pod) {
```
Comment (adambkaplan) [inline pkg/build/controller/strategy/util.go]: No longer needed - @nalind laid the groundwork in prior commits.

**Comment by adambkaplan** [comment]
/retest

centos mirror flake

**Comment by bparees** [comment]
/lgtm

but depending on what this does, you may be breaking some tests that rely on the default registry search behavior that comes from the registries.conf file embedded in the image.  I couldn't tell if this PR is going to result in always mounting over that file w/ the configmap content, even if the user didn't supply config?

**Comment by adambkaplan** [comment]
@bparees this should no-op in the same fashion as #21653. We aren't overwriting the default registry config because we're mounting in a separate path. Per @nalind buildah will fall back to the image defaults if the `registries.conf` file does not exist.

**Pull request merged**